code/modules/lighting/lighting_object.dm

```diff
@@ -8,6 +8,9 @@
 	///the turf that our light is applied to
 	var/turf/affected_turf
 
+	/// Area which gets linked to a lighting object to make it consider the luminosity from the day/night blending from the area. Yes this isn't ideal, but applying luminosity up to 2 (from both sources) on the turf is not ideal either
+	var/area/day_night_area
```

Comment:
not a massive fan of how this works I'ma be honest, needing to reach in and touch the object, the datum var read, etc. sucks
Would a sources list/bitflag work here?

Reply:
I also don't like how this is tied directly to day night stuff by name, that's its purpose in terms of this pr but this allows an area to decide if a lighting object should provide lumin or not, it's more general then the name implies

Reply:
Do you have a way to implement the bitflag? I could set the bitflag and unset it accordingly. But it would just be two options

Reply:
I think I thought incorrectly about how overlay lighting worked.
Something like a bool or a flag seems less... messy? idk.
If the area's lum changes that's fine, can't be helped. I guess I'd mostly just object to the name being so bespoke if so.

I am a bit worried about what happens if areas change, or if a turf loses its lighting object, but that's not related to this all that directly